backend/internal/flow/utils/utils.go

```diff
@@ -286,15 +287,19 @@ func GetExecutorByName(execConfig *model.ExecutorConfig) (model.ExecutorInterfac
 }
 
 // getIDP retrieves the IDP by its name. Returns an error if the IDP does not exist or if the name is empty.
-func getIDP(idpName string) (*idpmodel.IDP, error) {
+func getIDP(idpName string) (*idpmodel.IdpDTO, error) {
```

Comment:
We should eventually get rid of retrieving IDP through a util.. Instead, we should inject the idpservice to the flow service and handle this directly in the flow service leve.